internal/services/tesla_fleet_api_service.go

```diff
@@ -247,6 +269,56 @@ func (t *teslaFleetAPIService) RefreshToken(ctx context.Context, refreshToken st
 	return tokResp, nil
 }
 
+var fields = TelemetryFields{
+	"ChargeState":         {IntervalSeconds: 300},
+	"Location":            {IntervalSeconds: 10},
+	"OriginLocation":      {IntervalSeconds: 300},
+	"DestinationLocation": {IntervalSeconds: 300},
+	"DestinationName":     {IntervalSeconds: 300},
+	"EnergyRemaining":     {IntervalSeconds: 300},
+	"VehicleSpeed":        {IntervalSeconds: 60},
+	"Odometer":            {IntervalSeconds: 300},
+	"EstBatteryRange":     {IntervalSeconds: 300},
+	"Soc":                 {IntervalSeconds: 300},
+	"BatteryLevel":        {IntervalSeconds: 60},
+}
+
+func (t *teslaFleetAPIService) SubscribeForTelemetryData(ctx context.Context, token, region, vin string) error {
+	baseURL := fmt.Sprintf(t.Settings.TeslaFleetURL, region)
+	u, err := url.JoinPath(baseURL, "/api/1/vehicles/fleet_telemetry_config")
+	if err != nil {
+		return err
+	}
+
+	exp := time.Now().AddDate(0, 0, 364).Unix()
```

Comment:
From [the docs](https://developer.tesla.com/docs/fleet-api#fleet_telemetry_config-create)

> If not specified, config will be valid indefinitely.

Maybe we do this?

We will need a way for them to delete later, I suppose.